executor/slow_query.go

```diff
@@ -663,7 +663,20 @@ func (e *slowQueryRetriever) parseLog(ctx context.Context, sctx sessionctx.Conte
 func (e *slowQueryRetriever) setColumnValue(sctx sessionctx.Context, row []types.Datum, tz *time.Location, field, value string, checker *slowLogChecker, lineNum int) bool {
 	factory := e.columnValueFactoryMap[field]
 	if factory == nil {
-		return true
+		switch {
+		// Fix issue 34320, when slow log time is not in the output columns, the time filter condition is mistakenly discard.
+		case field == variable.SlowLogTimeStr && checker != nil:
```

Comment:
The `user` column maybe has the same problem, neet also handle `user` column here.

Reply:
No, the logic for `time` column and `user` column is different, the predication conditions on `time` column will be pushed down to the `MemTable`, while the predication conditions on the other columns use the `Selection` executor, so they don't have the save problem. @crazycs520

Reply:
oh, yes